libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -889,6 +916,22 @@ void AP_Baro::update(void)
         }
     }
 
+    const uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - _field_elevation_last_ms >= 1000 && fabs(_field_elevation_active-_field_elevation) > 1.0) {
+      if (!AP::arming().is_armed()) {
+        _field_elevation_last_ms = now_ms;
+        _field_elevation_active = _field_elevation;
+        AP::ahrs().resetHeightDatum();
+        update_calibration();
+        BARO_SEND_TEXT(MAV_SEVERITY_INFO, "Barometer Field Elevation Set: %.0fm",_field_elevation_active);
+      }
+      else {
+        _field_elevation.set_and_save(_field_elevation_active);
```

Comment:
```suggestion
        _field_elevation.set(_field_elevation_active);
```
No point in saving since we reset a boot anyway.

Reply:
Was following the template of temperature but yeah valid point

Reply:
Updated with suggested commit